rockcraft.yaml

```diff
@@ -118,6 +118,7 @@ parts:
       - python3-pip
     overlay-script: |
       mkdir -p $CRAFT_PART_INSTALL/opt/spark8t/python/dist
+      pip install --upgrade pip setuptools
```

Comment:
as discussed please pin to specific versions of `setuptools` and `pip`

Reply:
Uhm, I had a bit of closer look to this. The setuptools-59 would still be in the image, as this is part of the ubuntu base image. Indeed, that version does not get uninstalled as it is under `/usr`, in the logs I find

```
Attempting uninstall: setuptools
    Found existing installation: setuptools 59.6.0
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'setuptools'. No files were found to uninstall.
```

That installation comes from Ubuntu, which is up-to-date, i.e. in a multipass Ubuntu image I do get:

```
ubuntu@my-test-sandbox:~$ sudo apt-get upgrade python3-setuptools
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-setuptools is already the newest version (59.6.0-1.2ubuntu0.22.04.1).
Calculating upgrade... Done
The following packages have been kept back:
  linux-headers-generic linux-headers-virtual linux-image-virtual linux-virtual
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.
```

So it is probably all right to keep that setuptools version which will hopefully be bumped once a new base image will be released. 

I'm therefore inclined to remove these upgrades from this PR, and rely on the ones provided in the base